lms/djangoapps/instructor/views/util_test_grades.py

```diff
@@ -0,0 +1,136 @@
+import csv
+import json
+from datetime import datetime
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from pytz import UTC
+
+from student.models import CourseEnrollment
+from lms.djangoapps.grades.new.course_grade_factory import CourseGradeFactory
+from courseware import courses
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from openedx.core.djangoapps.content.course_structures.models import CourseStructure
+
+
+class ServiceGrades(object):
+
+    def __init__(self, course_id):
+        self.course_key = CourseKey.from_string(course_id)
+        self.course = courses.get_course_by_id(self.course_key)
+        self.students = CourseEnrollment.objects.users_enrolled_in(self.course_key)
+
+    def get_grades(self):
+        course_grades = []
+        for student in self.students:
+            course_grade_factory = CourseGradeFactory().create(student, self.course)
+            gradeset = course_grade_factory.summary
+            gradeset["username"] = course_grade_factory.user
+            gradeset["fullname"] = "{} {}".format(student.first_name, student.last_name)
+            course_grades.append(gradeset)
+
+        return course_grades
+
+    def by_section(self):
+        course_grade = self.get_grades()
+        score_by_section = []
+        initial_header_rows = ['username']
+        for student in course_grade:
+            course_grade_factory = CourseGradeFactory().create(student["username"], self.course)
+            sections = course_grade_factory.chapter_grades
+            section_dict = {}
+            for section in sections.items():
+                section_dict["username"] = course_grade_factory.user.username
+                score = course_grade_factory.score_for_chapter(section[0])
+                # Score object is a tuple: (earne, possible). We need only get earned value.
+                # import ipdb; ipdb.set_trace()
+                section_dict[section[1]["display_name"]] = score[0]
+                initial_header_rows.append(section[1]["display_name"])
+
+            score_by_section.append(section_dict)
+
+        # We need to remove repeated values in this array, due to in the loop above
+        # is repeated for each user
+        header_rows = list(set(initial_header_rows))
+        self.build_csv('section_report.csv', header_rows, score_by_section)
+
+        return score_by_section
+
+    def by_assignment_type(self):
+        course_grade = self.get_grades()
+        structure_course = CourseStructure.objects.get(course_id=self.course_key).structure_json
+        structure_course = json.loads(structure_course)
+
+        headers = []
+        assignment_type_grades = []
+
+        # We use by_section method to get grades and chapters names and add it to the headers
+        # but first we need to proccess this result deleting the username key, due values of usernames
+        # will be the rows and not the headers.
+        chapters = self.by_section()
```

Comment:
@diegomillan Note that here we are using our by_section method to get access to section grades, but at the same time by_section generates a CSV (line 57), so when we reuse it in by_assignmen_type method we are generating again a CSV for by section report. The only thing that I think could solve it would have another method exclusively to create the report only asking as parameter the result of by_section or by_assignment_type and this new method call build_csv. What do you think?